diff --git a/src/include/processor/operator/order_by/order_by_key_encoder.h b/src/include/processor/operator/order_by/order_by_key_encoder.h
index 3ca92de01..51ace06c5 100644
--- a/src/include/processor/operator/order_by/order_by_key_encoder.h
+++ b/src/include/processor/operator/order_by/order_by_key_encoder.h
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <cstring>
+
 #include <functional>
 #include <vector>
 
@@ -54,13 +56,17 @@ public:
     static uint32_t getNumBytesPerTuple(const std::vector<common::ValueVector*>& keyVectors);
 
     static inline uint32_t getEncodedFTBlockIdx(const uint8_t* tupleInfoPtr) {
-        return *(uint32_t*)tupleInfoPtr;
+        uint32_t index;
+        std::memcpy(&index, tupleInfoPtr, sizeof(index));
+        return index;
     }
 
     // Note: We only encode 3 bytes for ftBlockOffset, but we are reading 4 bytes from tupleInfoPtr.
     // We need to do a bit mask to set the most significant byte to 0x00.
     static inline uint32_t getEncodedFTBlockOffset(const uint8_t* tupleInfoPtr) {
-        return (*(uint32_t*)(tupleInfoPtr + 4) & 0x00FFFFFF);
+        uint32_t offset;
+        std::memcpy(&offset, tupleInfoPtr + 4, sizeof(offset));
+        return offset & 0x00FFFFFF;
     }
 
     static inline uint8_t getEncodedFTIdx(const uint8_t* tupleInfoPtr) {
diff --git a/src/processor/operator/order_by/order_by_key_encoder.cpp b/src/processor/operator/order_by/order_by_key_encoder.cpp
index a31064eb5..04f320cc9 100644
--- a/src/processor/operator/order_by/order_by_key_encoder.cpp
+++ b/src/processor/operator/order_by/order_by_key_encoder.cpp
@@ -182,8 +182,8 @@ void OrderByKeyEncoder::encodeFTIdx(uint32_t numEntriesToEncode, uint8_t* tupleI
         auto nextBatchOfEntries = std::min(numEntriesToEncode - numUpdatedFTInfoEntries,
             numTuplesPerBlockInFT - ftBlockOffset);
         for (auto i = 0u; i < nextBatchOfEntries; i++) {
-            *(uint32_t*)tupleInfoPtr = ftBlockIdx;
-            *(uint32_t*)(tupleInfoPtr + 4) = ftBlockOffset;
+            std::memcpy(tupleInfoPtr, &ftBlockIdx, sizeof(ftBlockIdx));
+            std::memcpy(tupleInfoPtr + 4, &ftBlockOffset, sizeof(ftBlockOffset));
             *(uint8_t*)(tupleInfoPtr + 7) = ftIdx;
             tupleInfoPtr += numBytesPerTuple;
             ftBlockOffset++;
